services/lgtm/lgtm-grade.service.js

```diff
@@ -1,99 +1,88 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const { checkErrorResponse } = require('../../lib/error-helper')
-
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class LgtmGrade extends LegacyService {
-  static get category() {
-    return 'analysis'
-  }
+const LgtmBaseService = require('./lgtm-base')
 
+module.exports = class LgtmGrade extends LgtmBaseService {
   static get route() {
     return {
       base: 'lgtm/grade',
+      pattern: ':language/g/:user/:repo',
+    }
+  }
+
+  async handle({ language, user, repo }) {
+    const data = await this.fetch({ user, repo })
+
+    if (!('languages' in data)) {
+      return this.constructor.render({
+        grade: 'invalid response data',
+        color: 'red',
+      })
```

Comment:
If you make `languages` required in the schema, we can get rid of this block - `validate()` will throw the error for us